File: Week3-4/Section 2/fib.js
function fibonacciIterative(n) {
  var _a;
  if (n <= 1) {
    return n;
  }
  var a = 0,
    b = 1;
  for (var i = 2; i <= n; i++) {
    (_a = [b, a + b]), (a = _a[0]), (b = _a[1]);
  }
  return b;
}
function main() {
  console.log("TypeScript Fibonacci:");
  console.log("Iterative fib(10): ".concat(fibonacciIterative(10)));
}
main();

File: Week3-4/Section 2/fib.py
def fibonacci_iterative(n):
    """Calculate Fibonacci number using iteration (more efficient)"""
    if n <= 1:
        return n
    
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

def main():
    print("Python Fibonacci:")
    print(f"Iterative fib(10): {fibonacci_iterative(10)}")
        

if __name__ == "__main__":
    main()